### ISSUE TITLE
E/CustomActivityOnCrash: App has crashed, executing CustomActivityOnCrash's UncaughtExceptionHandler

### ISSUE THREAD

**Issue by utkarshGithub** [Ereza/CustomActivityOnCrash#44]
I am using custom error activity , below is  my manifest file 
 ```
<activity
            android:name="com.o1.shop.ui.activity.CustomErrorActivity"
            android:process=":error_activity"
            android:screenOrientation="portrait">
            <intent-filter>
                <action android:name="cat.ereza.customactivityoncrash.ERROR"/>
            </intent-filter>
        </activity>
```

and this my application file 

```
CleverTapAPI.changeCredentials(DeviceUtils.getCleverTapAccountID(BuildConfig.FLAVOR), DeviceUtils.getCleverTapAccountToken(BuildConfig.FLAVOR));
        ActivityLifecycleCallback.register(this);
        super.onCreate();
        CaocConfig.Builder.create()
                .enabled(true)
                .minTimeBetweenCrashesMs(2000) //default: 3000
                .errorActivity(CustomErrorActivity.class) //default: null (default error activity)
                .apply();
```
i cannot see the custom crash activity

**Comment by Ereza** [comment]
Hi @utkarshGithub, sorry for taking a while to reply. Did you manage to solve the problem?

Your code appears to be fine (however, if you set the errorActivity via CaocConfig, the intent-filter is not really needed). Could you provide a minimal project so I can check if there is any other issue that causes this?

Thanks!

**Comment by DominuS-RU** [comment]
The same issue. Nothing happens after force crash at the end of onCreate in activity

**Comment by Ereza** [comment]
Hi @DominuS-RU, can you provide a minimal project with all the code so I can reproduce the issue? I'm assuming that the problem doesn't happen on the sample project, can you check?

Thanks!!

**Comment by DominuS-RU** [comment]
@Ereza Hello! 
I've updated my previous comment with some details. 
I'll test sample project and will try to give you test one in some hours.

Hope, my current details will help you.

**Comment by DominuS-RU** [comment]
@Ereza, I've found the mistake. 
It's the config for your sample:
```
CaocConfig.Builder.create()
.backgroundMode(CaocConfig.BACKGROUND_MODE_SILENT)
.apply();
```
An exception for MainActivity
`throw new RuntimeException("boom");`

![studio64_2018-01-19_16-58-20](https://user-images.githubusercontent.com/10329086/35154117-ffe83328-fd39-11e7-897a-45b744fd616a.png)

Then, start you application and see nothing.
The main issue in .backgroundMode(CaocConfig.BACKGROUND_MODE_SILENT), if we delete this one the library will work fine. All behavior is repeated as I've described in my first comment.

**Comment by DominuS-RU** [comment]
Anything new about this issue?

**Comment by puxiaoshuai** [comment]
App has crashed, executing CustomActivityOnCrash's UncaughtExceptionHandler
what's wrong?

**Comment by Yexingshuai** [comment]
I had the same problem，i cannot see the custom crash activity

**Comment by Ereza** [comment]
Hi, sorry for the ultralate reply.

With the examples you provided, I saw that there was an issue when the background mode is "silent" and the app is being launched. The background/foreground detection mechanism relies on **started** activities, and obviously a new activity that crashes in `onCreate` is not started yet, so the library thinks that the app is in background and does not show the error activity.

I solved this by recording the last time an activity was created (`onActivityCreated` is called before `onCreate` ends). If the activity crashes while in "background" but the last activity creation was less than 500 ms ago, the app is assumed to be in foreground and the activity is shown.

This will be available in the next release of the library.
